go.mod

```diff
@@ -30,6 +30,8 @@ require (
 	github.com/openshift/machine-config-operator v0.0.1-0.20230330142923-2832f049b3f4
 	github.com/openshift/operator-custom-metrics v0.5.1
 	github.com/openshift/osde2e-common v0.0.0-20240531074950-36a7055798ae
+	// Note: Please revisit the stop-gap in ./main.go > cleanupLockForLeader when updating operator-lib module.
+	// If the issue has been fixed upstream, the stop-gap should be removed.
```

Comment:
Thanks Tafhim!
These are great!

Reply:
@Tafhim - I think it's better to have such notes added in markdown or the code pkg files for reference. However, if we want to ensure that a go.mod version is tied to version due to a known issue etc and we don't accidentally update it then it might be better to have it commented with the upstream issue/bug as follows under the `replace` directive. Example:
```
replace (
    github.com/operator-framework/operator-lib v0.12.0 // Refer upstream issue <link>
)
```
It would also be fine to have a "maintenance" section added in this repo's docs - https://github.com/openshift/managed-upgrade-operator/blob/master/docs/development.md 

We can have a separate maintenance.md file covering overall process and highlight this dependency requirement/reason there.

Reply:
I suggested putting it here so it would be noticed if the version was updated.
As far as I am aware we don't have the link for the upstream issue yet so a more generic note was suggested.
Putting this somewhere it's not going to be noticed may mean the patched release will happen without anyone noticing and we will have  two services updating the same resource which may cause issues.